### ISSUE TITLE
After recent updates the Theme initial data is not there

### ISSUE THREAD

**Issue by MounirMesselmeni** [fabiocaccamo/django-admin-interface#229]
**Python version**
3.10

**Django version**
4.1.4

**Package version**
latest

**Current behavior (bug description)**
Theme.objects.active returning None, could be has something todo with the removal of post_migrate signal

**Expected behavior**
Works as before or some docs about how and when data is loaded initially.

**Comment by fabiocaccamo** [comment]
@merwok @VaZark @zN3utr4l here we are :)

**Comment by fabiocaccamo** [comment]
@MounirMesselmeni have you upgraded the library or you have installed it for the first time right now?

**Comment by merwok** [comment]
Two venues in my opinion:
- add a migration to create a Theme with all current options
- instruct people to run a loaddata command with one of the themes then edit it

**Comment by VaZark** [comment]
Is there anywhere we use theme.objects directly instead of get_active_theme? Ideally we can reuse that logic for initial load

**Comment by fabiocaccamo** [comment]
Before doing any change I would like to:

- [ ] replicate the error

- [ ] write a failing test and then fix it

- [ ] write a fix for make the test pass

_(today is the checkboxes day)_

**Comment by MounirMesselmeni** [comment]
Ok, I do use Theme.objects to override the theme based on the environment and some variables. The flow is run migrate and then run the custom management command which does not use get_active_theme btw if that one is populating data into the database on the first run.

**Comment by MounirMesselmeni** [comment]
@fabiocaccamo i did upgrade but the issue is only when using a fresh database.

**Comment by fabiocaccamo** [comment]
@MounirMesselmeni what do you mean with:

> Theme.objects.active returning None

- Do you access the theme programmatically?

- Do you see any error?

- Do you go to the admin and see no themes?

- Is the admin styled with the custom default theme?

**Comment by MounirMesselmeni** [comment]
If it helps this is what I'm doing:

```python
def setup_theme(self, env: str) -> None:
        base_title = package_settings.APP_NAME
        theme_settings = {
            'css_module_rounded_corners': True,
            'env_name': env.upper(),
            'title': f'{base_title} ({env.upper()})',
        }
        env_themes = package_settings.ENV_THEMES
        #theme: Theme = Theme.objects.filter(active=True).first()
        theme: Theme = Theme.get_active_theme()
        theme_settings.update(env_themes[env.lower()])
        for field_name, field_value in theme_settings.items():
            setattr(theme, field_name, field_value)
        theme.save()
        del_cached_active_theme()
```

**Comment by MounirMesselmeni** [comment]
Thank you all for the quick answers and your help! Appreciated!

**Comment by fabiocaccamo** [comment]
Maybe we can prent this kind of problems by using a custom manager.

**Comment by merwok** [comment]
Definitely, Theme should use a ThemeManager with custom methods on it, not classmethods.  That’s the django way!
